mono/sgen/sgen-gc.c

```diff
@@ -1802,6 +1802,12 @@ collect_nursery (const char *reason, gboolean is_overflow)
 	TV_GETTIME (btv);
 	time_minor_scan_roots += TV_ELAPSED (atv, btv);
 
+	SGEN_LOG (2, "Minor scan roots: %lld usecs (accumulated major scan=%lld, accumulated los scan=%lld, scan jobs=%d)",
+		(long long)(TV_ELAPSED (atv, btv) / 10),
```

Comment:
The scan roots time looks inconsistent. For serial it is the time it takes to scan registered roots, while on parallel it is the time it takes for all jobs to finish (including card table scanning)

Reply:
Yes, similar problem for several of the timers and logging within this part since in parallel the enqueue_scan_remembered_set_jobs will be included in different accumulated timers compared to not running parallel. In parallel it will be hard to separate senqueue_scan_remembered_set_jobs and enqueue_scan_from_roots_jobs.

Reply:
Maybe we should exclude some of the incorrect loggings in parallel, like the scan old geneneration and instead have one logging in the end that will give info around both scan remset and scan roots with the accumulated info for major/los scan root jobs as well. In that case the time_minor_scan_remsets and time_minor_scan_roots are also incorrect so maybe we shouldn't update those counters when running in parallel?

Reply:
I have no opinion about this. Feel free to display whatever logs you feel it would be useful. In this PR you suggested to log scan roots, scan major and scan los. We already have the cumulated times for scan major and scan los, you can add the correct accumulated time for each root job and you can display the same thing. For example